src/macaron/slsa_analyzer/provenance/intoto/v01/__init__.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module handles in-toto version 0.1 attestations."""
+
+from __future__ import annotations
+
+from typing import TypedDict, TypeGuard
+
+from macaron.slsa_analyzer.provenance.intoto.errors import ValidateInTotoPayloadError
+from macaron.util import JsonType
+
+
+class InTotoStatement(TypedDict):
+    """An in-toto version 0.1 statement.
+
+    This is the type of the payload in an in-toto version 0.1 attestation.
+    Specification: https://github.com/in-toto/attestation/tree/main/spec/v0.1.0#statement.
+    """
+
+    _type: str
+    subject: list[InTotoSubject]
+    predicateType: str  # noqa: N815
+    predicate: dict[str, JsonType] | None
+
+
+class InTotoSubject(TypedDict):
+    """An in-toto subject.
+
+    Specification: https://github.com/in-toto/attestation/tree/main/spec/v0.1.0#statement.
+    """
+
+    name: str
+    digest: dict[str, str]
+
+
+def validate_intoto_statement(payload: dict[str, JsonType]) -> TypeGuard[InTotoStatement]:
+    """Validate the statement of an in-toto attestation.
+
```

Comment:
Same comment regarding adding TODO and link this [issue](https://github.com/oracle/macaron/issues/426).

Reply:
See d2b1f228d8c38248bf71cb1e7788011bf09c6e6c.